### PR TITLE
OrderFilter: Allow to set default order for property

### PR DESCRIPTION
**Pull request by ogizanagi**

### CODE REVIEW

**Comment by dunglas** [comment]
Great! Can you run PHP CS Fixer please?

**Comment by ogizanagi** [comment]
Done.

File: Doctrine/Orm/Filter/OrderFilter.php [hunk 1]
```diff
@@ -55,10 +55,15 @@ public function apply(ResourceInterface $resource, QueryBuilder $queryBuilder, R
         $fieldNames = array_flip($this->getClassMetadata($resource)->getFieldNames());
 
         foreach ($properties as $property => $order) {
-            $order = strtoupper($order);
+            if ($this->isPropertyEnabled($property) && isset($fieldNames[$property])) {
```
Comment (dunglas) [inline Doctrine/Orm/Filter/OrderFilter.php]: You can reduce the cyclomatic complexity by using a `continue` statement:

```
if (!$this->isPropertyEnabled($property) || !isset($fieldNames[$property])) {
    continue;
}
```

File: Doctrine/Orm/Filter/OrderFilter.php [hunk 2]
```diff
@@ -55,10 +55,15 @@ public function apply(ResourceInterface $resource, QueryBuilder $queryBuilder, R
...
+                if ('' === $order && isset($this->properties[$property])) {
+                    $order = $this->properties[$property];
+                }
+                $order = strtoupper($order);
```
Comment (dunglas) [inline Doctrine/Orm/Filter/OrderFilter.php]: Add a blank line before and remove the one after.

**Comment by ogizanagi** [comment]
@dunglas : I also added tests, which allowed me to spot [this](https://github.com/dunglas/DunglasApiBundle/pull/108/files#diff-1dcf7839408dce90e2604a443461dee3R422).
As you can see, php `http_build_query` function reintroduced the param with an extra `=` despite it has no value. Does it matter ? This seems to be valid according to the [URI RFC](http://tools.ietf.org/html/rfc3986#section-3.4)

**Comment by dunglas** [comment]
Looks OK for me.

**Comment by dunglas** [comment]
Can you squash your commits?

**Comment by ogizanagi** [comment]
Sorry, squashed.

**Pull request merged**